# Exercise 4.2 — A Simple Climate Model

In this exercise, we explore a basic **radiative energy balance model** for a planet, describing how incoming solar radiation and outgoing thermal radiation control the planet's temperature.

The governing equation is:

$
c \frac{dT}{dt} = SW_{\downarrow} - LW_{\uparrow}
$

Where:
- **$ T $** is the globally averaged surface temperature (K).
- **$ c $** is the effective global heat capacity per unit area.
- **$ SW_{\downarrow} $** is the incoming shortwave (solar) radiation.
- **$ LW_{\uparrow} $** is the outgoing longwave (thermal infrared) radiation.

**Associated physical laws:**

- **Incoming solar radiation** (assuming uniform distribution over a spherical planet's surface):

$
SW_{\downarrow} = \frac{S_0}{4}(1 - \alpha_p)
$

Where:
- $ S_0 $ is the solar constant (incoming solar flux at the top of atmosphere, [W/m²]).
- $ \alpha_p $ is the planet’s albedo (fraction of incoming radiation reflected back to space).

- **Outgoing longwave radiation** (Stefan-Boltzmann law for thermal radiation):

$
LW_{\uparrow} = \epsilon \sigma T^4
$

Where:
- $ \epsilon $ is the surface emissivity (dimensionless, typically ≤ 1).
- $ \sigma $ is the Stefan-Boltzmann constant $ (5.67 \times 10^{-8}\, W\, m^{-2}\, K^{-4}) $.

---

# Exercise 4.3 — Equilibrium Temperature

At equilibrium, energy in equals energy out:

$
SW_{\downarrow} = LW_{\uparrow}
$

Solving for **$ T_{eq} $**:

$
T_{eq} = \left( \frac{S_0 (1 - \alpha_p)}{4 \epsilon \sigma} \right)^{1/4}
$

We’ll compute this for **Mercury**, **Venus**, and **Earth** (assuming emissivity $ \epsilon = 1 $).

In [ ]:
# Constants
sigma = 5.67e-8  # Stefan-Boltzmann constant (W/m²/K⁴)
epsilon = 1      # assuming perfect blackbody for simplicity

# Solar constant (W/m²) from https://de.wikipedia.org/wiki/Solarkonstante
S0_earth = 1361       # Earth
S0_mercury = 9126     # Mercury
S0_venus = 2601       # Venus

# Albedos (dimensionless)
albedo_earth = 0.3  # https://de.wikipedia.org/wiki/Albedo
albedo_mercury = 0.12  # https://eph.univie.ac.at//Planetologie/Merkur/index.htm
albedo_venus = 0.76  # https://de.wikipedia.org/wiki/Albedo


# Equilibrium temperature formula
def equilibrium_temperature(S0, albedo, epsilon=1):
    return ((S0 * (1 - albedo)) / (4 * epsilon * sigma))**0.25

# Calculate temperatures
T_earth = equilibrium_temperature(S0_earth, albedo_earth)
T_mercury = equilibrium_temperature(S0_mercury, albedo_mercury)
T_venus = equilibrium_temperature(S0_venus, albedo_venus)

# Print results
print(f"Earth: {T_earth:.2f} K")
print(f"Mercury: {T_mercury:.2f} K")
print(f"Venus: {T_venus:.2f} K")

Earth: 254.58 K
Mercury: 433.79 K
Venus: 229.05 K


### Comparison of Calculated Equilibrium Temperatures with Actual Surface Temperatures

The calculated equilibrium temperatures (without greenhouse effect) for the planets are:

- **Earth:** 254.6 K
- **Mercury:** 433.8 K
- **Venus:** 229.0 K

According to [Wikipedia: List of Solar System Planets](https://en.wikipedia.org/wiki/List_of_Solar_System_planets), the observed surface temperatures (in Kelvin) are:

| Planet   | Min (K) | Mean (K) | Max (K) |
|----------|---------|----------|---------|
| Mercury  | 100     | 440      | 780     |
| Venus    | 710     | 737      | 770     |
| Earth    | 213     | 288      | 331     |

**Comparison:**

- **Mercury:** The calculated temperature (433.8 K) is very close to the observed mean (440 K).
- **Venus:** The calculated temperature (229.0 K) is far below the observed mean (737 K), highlighting the strong greenhouse effect on Venus.
- **Earth:** The calculated temperature (254.6 K) is significantly below the actual mean (288 K), showing the natural greenhouse effect on Earth aswell as internal heating.

**Conclusion:**  
The simple radiative model underestimates the surface temperatures of Venus and Earth because it does not account for the greenhouse effect. For Mercury, which has almost no atmosphere, the model matches reality quite well.